examples/tests/cluster-addons/step2/index.ts

```diff
@@ -0,0 +1,55 @@
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+import * as iam from "./iam";
+
+// IAM roles for the node groups.
+const role = iam.createRole("cluster-addons");
+
+// Create a new VPC
+const eksVpc = new awsx.ec2.Vpc("cluster-addons", {
+  enableDnsHostnames: true,
+  cidrBlock: "10.0.0.0/16",
+});
+
+// Create an EKS cluster.
+const cluster = new eks.Cluster("cluster-addons", {
+  skipDefaultNodeGroup: true,
+  vpcId: eksVpc.vpcId,
+  authenticationMode: eks.AuthenticationMode.API,
+  publicSubnetIds: eksVpc.publicSubnetIds,
+  privateSubnetIds: eksVpc.privateSubnetIds,
+  vpcCniOptions: {
+    disableTcpEarlyDemux: true,
+    logLevel: "INFO",
+    securityContextPrivileged: true,
+    enableNetworkPolicy: true,
+    
+    // required for security groups for pods
+    enablePodEni: true,
+    configurationValues: {
+      env: {
+        POD_SECURITY_GROUP_ENFORCING_MODE: "standard",
+      }
+    },
+  },
+});
+
+// Export the cluster's kubeconfig.
+export const kubeconfig = cluster.kubeconfig;
+export const clusterName = cluster.core.cluster.name;
+
+const ng = eks.createManagedNodeGroup("cluster-addons", {
```

Comment:
Out of scope of this PR - but remind me do we like this API that creates an aws.eks.NodeGroup instance vs some form of API that's a wrapper Component resource e.g. `new eks.ManagedNodeGroup()`

Reply:
I'm indifferent about those methods tbh. The one shortcoming of them is that they're not documented in the registry (which we could do if we really wanted to).

Reply:
Yeah definitely more of a style/consistency issue than anything.